lib/Runtime/Debug/TTExecutionInfo.cpp

```diff
@@ -896,7 +896,8 @@ namespace TTD
                 //we moved to a new statement
                 Js::FunctionBody::StatementMap* pstmt = fb->GetStatementMaps()->Item(cIndex);
                 bool newstmt = (cIndex != cfinfo.CurrentStatementIndex && pstmt->byteCodeSpan.begin <= (int)bytecodeOffset && (int)bytecodeOffset <= pstmt->byteCodeSpan.end);
-                if (newstmt)
+                bool startUnaligned = (cfinfo.CurrentStatementIndex == -1) && (pstmt->byteCodeSpan.begin != (int)bytecodeOffset); //make sure async step back is ok
```

Comment:
Nit: I mildly prefer testing against the positive case rather than testing the inverse of the negative case (i.e have a variable called startAligned instead).

What is expected to happen if the statement is not aligned?

I'm also unclear as to the circumstances when pstmt->byteCodeSpan.begin != byteCodeOffset- I can believe it happens but I'm not sure why it's related to async calls- can you add a clarifying comment here that speaks to that?